src/Domain/DonationRepository.php

```diff
@@ -183,6 +184,10 @@ public function allocateMatchFunds(Donation $donation, bool $mistrustOrmFundingC
 
         $this->getEntityManager()->flush();
 
+        // After new funds are allocated, we want releasing them to be allowed without waiting for lock expiry, in case
+        // e.g. a donor changes their donation amount for a second time within that window.
+        $this->getFundsReleaseLock($donation)->release();
```

Comment:
confused about why we seem to be acquiring the lock and releasing it on the same line - shouldn't we have a variable for a lock aquired earlier that we're now ready to release.

Reply:
I was thinking about amount changes, which means the release and allocation are different API calls so the lock isn't in memory. However I also forgot that that means the amended version of the donation is a _new_ donation; I don't think we patch core amounts. That being the case, maybe this is redundant. I can't immediately think of another reason we'd need to release, allocate and release again within 5 minutes for the same donation.

Reply:
Yes we don't allow changes to donation amount on any given donation. https://github.com/thebiggive/matchbot/blob/bc124b3e2bbf6d8ef513fd05eb16aea06f7b37eb/src/Application/Actions/Donations/Update.php#L251